WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift

```diff
@@ -243,8 +243,8 @@ private extension AddProductCoordinator {
         viewModel.onProductCreated = { [weak self] product in
             guard let self else { return }
             self.onProductCreated(product)
-            if self.isFirstProduct, let url = URL(string: product.permalink) {
-                self.showFirstProductCreatedView(productURL: url,
+            if self.isFirstProduct {
+                self.showFirstProductCreatedView(product: product,
                                                  showShareProductButton: viewModel.canShareProduct())
```

Comment:
Unrelated: should `canShareProduct` also check for `product.permalink` just in case?

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/9875/commits/1d3bc75819f0cacfe0a65776540703f6ec416292 and https://github.com/woocommerce/woocommerce-ios/pull/9875/commits/a2613eeba141805b802cee38124f969985ecf914. Unit tests are added/updated in subsequent commits.